src/main/java/wansbot/WansBot.java

```diff
@@ -0,0 +1,337 @@
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        }
+    }
+
+    /**
+     * Throws InputEmptyException when user does not input a deadline after "deadline /by".
+     */
+    private static void checkMissingInputDeadline(String userInput) {
+        String[] splitUser = userInput.split(" /by ", 2);
+        if (splitUser.length < 2) {
+            throw new InputEmptyException(userInput, "/by");
+        }
+    }
+
+    /**
+     * Throws custom InputEmptyException for events when either missing from date after /from, missing to date after
+     * /to or missing both.
+     */
+    protected static void missingInputEvent(String userInput) {
+        String[] splitUserStartDate = userInput.split(" /from ", 3);
+        if (splitUserStartDate.length < 2) {
+            throw new InputEmptyException(userInput, "/from");
+        }
+        String[] splitUserEndDate = splitUserStartDate[1].split(" /to ", 2);
+        if (splitUserEndDate.length < 2) {
+            throw new InputEmptyException(userInput, "/to");
+        }
+    }
+
+    /**
+     * Handles marking tasks function by specifying how caught exceptions are dealt with. Prints to console what
+     * user needs to correct for the command to work.
+     */
+    private String markTasks(String userInput) {
+
+        try {
+            checkNumInput(userInput, userTaskList.numOfTasks());
+            int posTask = Integer.parseInt(userInput.substring(5)) - 1;
+            userTaskList.getTask(posTask).finish();
+            return ui.handleSuccesfulMarking(userTaskList, posTask);
+        } catch (NumberFormatException e) {
+            return ui.handleMarkingFormat();
+        } catch (NotANumMarkingException e) {
+            return ui.handleInvalidNum();
+        }
+    }
+
+    /**
+     * Handles unmarking tasks function by specifying to user what went wrong.
+     */
+    private String unmarkTasks(String userInput) {
+        try {
+            checkNumInput(userInput, userTaskList.numOfTasks());
+            int posTask = Integer.parseInt(userInput.substring(7)) - 1;
+            userTaskList.getTask(posTask).unfinish();
+            return ui.handleSuccesfulUnmarking(userTaskList, posTask);
+        } catch (NumberFormatException e) {
+            return ui.handleUnmarkingFormat();
+        } catch (NotANumMarkingException e) {
+            return ui.handleInvalidNum();
+        }
+    }
+
+    /**
+     * Adds Todos to the userTaskList.
+     */
+    private String addTodos(String userInput) {
+        Todos newTodo = new Todos(userInput.substring(5));
+        userTaskList.add(newTodo);
+        return ui.handleSuccessfulAdd(newTodo);
+    }
+
+    /**
+     * Adds Deadlined to userTaskList. If there is invalid input, WansBot will guide user on the correct commands.
+     */
+    private String addDeadlined(String userInput) {
```

Comment:
did you maybe mean to write addDeadlines here?

Reply:
Actually, because Deadlined is the name of my class, I intentionally made it that way. Maybe I should refactor it to Deadline to make it clearer